controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -305,30 +311,21 @@ func (r *StorageConsumerUpgradeReconciler) reconcileStorageConsumer(ctx context.
 	return nil
 }
 
-// getStorageRequestHash generates a hash for a StorageRequest based
-// on the MD5 hash of the StorageClaim name and storageConsumer UUID.
-func getStorageRequestHash(consumerUUID, storageClaimName string) string {
-	s := struct {
-		StorageConsumerUUID string `json:"storageConsumerUUID"`
-		StorageClaimName    string `json:"storageClaimName"`
-	}{
-		consumerUUID,
-		storageClaimName,
+// reconcileStorageClient Updates the status of the internal client to point to the new consumer
+func (r *StorageConsumerUpgradeReconciler) reconcileStorageClient(
+	ctx context.Context,
+	storageClientName string,
+	storageConsumerUID string,
+) error {
+	storageClient := &ocsclientv1a1.StorageClient{}
+	storageClient.Name = storageClientName
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(storageClient), storageClient); err != nil {
+		return err
 	}
 
-	requestName, err := json.Marshal(s)
-	if err != nil {
-		panic("failed to marshal storage class request name")
+	storageClient.Status.ConsumerID = storageConsumerUID
+	if err := r.Client.Status().Update(ctx, storageClient); err != nil {
+		return err
 	}
```

Comment:
The decision need to be made from the outside. 
If you are already inside a reconcile function you should reconcile